### ISSUE TITLE
PlayIntegrityFix v3.8 may be an issue to pass DEVICE_INTEGRITY (subject change from "/system" mounts RO roadblock?)

### ISSUE THREAD

**Issue by MaccabeeY** [chiteroman/PlayIntegrityFix#241]
Been trying to get a Pixel 5 with factory image (android 12; Magisk 26.4004-canary) to pass MEETS_DEVICE_INTEGRITY for a few days now with no success.  As I tried to dig in to see if I could find a/the hangup, I think I may have discovered a roadblock which, I think, is that the "/system" partition mounts RO so "build.prop" cannot be modified; and as yet I've not found a way to change that.

So am I right, no way to get a factory image to mount RW?  And, if so, no way DEVICE_INTEGRITY will never pass on a rooted factory image?

**Comment by MaccabeeY** [comment]
UPDATE:  I now have installed LineageOS 20 for Pixel 5 and carefully followed all instructions provided for PlayIntegrityFix plus TheFreeman193.  Have run "pickaprint.sh" at least 5 times and converted pickaprint json & put it in /data/adb/pif.json.

However, each time after "pickaprint.sh" is run (& all Play data clear & device rebooted), then after reboot, "TB Checker" is run & still reports the ORIGINAL "device build fingerprint" every time.  Wondering if chiteroman/PlayIntegrityFix is not working correctly since "TB Checker" always reports the original fingerprint even after "pickaprint.sh"?

**Comment by vladrevers** [comment]
The file buid.prop and should not change, because the values are not read directly from it. You can see the necessary parameters with the getprop command to see if they are applied. For example, the output of the getprop ro.boot.verifiedbootstate command should be green.

All the verification applications I've used have always shown my device data, including Json responses, but not the swiped ones.

To summarize, you're wrong in your guesses almost everywhere, and you just need to find a working fingerprint and other data for pif.json. I myself have a Pixel 7 and a factory image, I had a Pixel 4a before that.

**Comment by vladrevers** [comment]
@MaccabeeY about getprop, this only applies to the values ​​​​that are in services.sh and post-fa-data.sh:

- resetprop_if_diff ro.build.tags release-keys
- resetprop_if_diff ro.boot.warranty_bit 0
- resetprop_if_diff ro.vendor.boot.warranty_bit 0
- resetprop_if_diff ro.vendor.warranty_bit 0
- resetprop_if_diff ro.warranty_bit 0
- resetprop_if_diff ro.is_ever_orange 0
- resetprop_if_diff ro.build.type user
- resetprop_if_diff ro.debuggable 0
- resetprop_if_diff ro.secure 1
- resetprop_if_diff ro.boot.flash.locked 1
- resetprop_if_diff ro.boot.vbmeta.device_state locked
- resetprop_if_diff vendor.boot.verifiedbootstate green
- resetprop_if_diff ro.boot.verifiedbootstate green
- resetprop_if_diff ro.boot.veritymode enforcing
- resetprop_if_diff vendor.boot.vbmeta.device_state locked

The values ​​from pif.drift are used only for GMS, and it seems only for those related to droidguard, I don’t know where to see that they are exactly applied)

**Comment by MaccabeeY** [comment]
Thanks for confirming, I suspected those associated values and for clarifying they pif.json values don't get show via getprops.  Kinda a bummer there we don't know a tool or way to see the pif.json values applied!

vlad, from what you've said, I suppose I'll just have to go on faith and keep trying different "TheFreeman193" files 'til one works (tho the bummer is that it seems Google has some kind of "10k" 24 hour limit thingy, so I can only try a tiny few per day)

**Comment by markfm62** [comment]
Also, is your kernel a true stock one, or are you running something that's been modified?  If it's a modified kernel you may be bumping into Google's new blacklisting of some kernel names. (No pif.json will fix this issue as of right now)

If it's not a modified kernel, and you've done the "normal" things (Play Services and Play Store on your Magisk DenyList, set Enforce DenyList not active/off, Use Shamiko module (v1.0.1 given you have Magisk Canary/26404), then you're probably running into luck-of-the-draw.  Some people have had to run a PIF picker many times before they ended up on one that isn't banned (a fingerprint ban is separate from a kernel name blacklist)

**Comment by MaccabeeY** [comment]
Thanks for that thought.  I imagine custom kernels would be an important consideration.

I'm attempting to pass DEVICE_INTEGRITY specifically with LineageOS version 19.1 (2022-12-28) that can be found at [Google Pixel 5 (redfin)](https://lineageosroms.com/redfin/).
I'm not sure how to determine if the kernel that is on my Pixel 5 from that package is "standard" or modified/custom.

I, personally, have not modified any of the content provided by that image, installing precisely what's been provided by LineageOS unchanged.

I do not have any knowledge whether LineageOS uses a stock or modified kernels in that or other packages.

I chose LineageOS since I've read a quantity of folks claiming to recently having passed DEVICE_INTEGRITY with LineageOS images.

Now that you've mentioned it, I"m hoping LineageOS (and that package specifically) includes a "standard" kernel (I'd hate to have to start over from scratch).

**Comment by vladrevers** [comment]
@MaccabeeY run the commands "uname -r" or "uname -a", and look at their output, if it makes clear that this is a third-party kernel, then it won’t work (for example, you will see the name of the firmware, the author or something else, like the name of the release)

**Comment by MaccabeeY** [comment]
Thanks!  But the output not quite clear for me, look like stock-like stuff, but I'm not sure what to look for.
Here's the output from those commands, in case it means more to you:

uname -r  ==>  4.19.224-g3c01f924b7c4

uname -a ==>  Linux localhost 4.19.224-g3c01f924b7c4 #1 SMP PREEMPT Wed Dec 28 05:41:56 UTC 2022 aarch64

**Comment by markfm62** [comment]
A not-too-wild guess - Google likely knows that is a lineageos-related kernel.  I did a search on "4.19.224-g3c01f924b7c4" and it brought me to a LineageOS github page: https://gitlab.com/LineageOS/issues/android/-/issues/5252

**Comment by MaccabeeY** [comment]
You probably right, but in my build.prop there isn't "bramble" (like on that thread) or any other LineageOS "codename" or indicator that I can tell, so hopefully...
i.e.
ro.product.system.brand=Android
ro.product.system.device=generic
ro.product.system.manufacturer=Android
ro.product.system.model=mainline
ro.product.system.name=mainline

And thanks for the that thread link, seeming out key folks here might be in the midst of some heavy "sherlocking" to see if they can figure out way(s) to code past the current struggles with DEVICE_INTEGRITY.

I'll keep a close watch & keep trying those pifs.  But if you're right & they detect non-google kernels & if the good programmers workiing here can't get around it in the sooner than later, likely gonna really hurt orgs. like LineageOS.

Before I had put on LineageOS, I first put on factory w/"clean" rooted (passed basic; root check; & xposed check) and still no DEVICE_INTEGRITY joy so that's when I said hell try LineageOS.

Anyway, I have faith in chiteroman & TheFreeman193, so I think I'm gonna wait a short bit, see what they come up with, before trashing my setup on LineageOS and starting all the way over with factory.

If you learn anything new, shoot it my way & I'll do same...

**Comment by markfm62** [comment]
I flagged it in the other Issue (the general one on google banning kernels).  There is more than one callout of that particular kernel in the LineageOS area, and it appears that Google is explicitly blacklisting things where they can clearly figure out that it's a custom.

**Comment by markfm62** [comment]
@MaccabeeY  - Send me an IM on Telegram (Mark M)  and I can provide you one of my own json to try

**Comment by MaccabeeY** [comment]
@markfm62, Incredibly generous of you, will do.

**Comment by XdekHckr** [comment]
> UPDATE: I now have installed LineageOS 20 for Pixel 5 and carefully followed all instructions provided for PlayIntegrityFix plus TheFreeman193. Have run "pickaprint.sh" at least 5 times and converted pickaprint json & put it in /data/adb/pif.json.
> 
> However, each time after "pickaprint.sh" is run (& all Play data clear & device rebooted), then after reboot, "TB Checker" is run & still reports the ORIGINAL "device build fingerprint" every time. Wondering if chiteroman/PlayIntegrityFix is not working correctly since "TB Checker" always reports the original fingerprint even after "pickaprint.sh"?

if you use tbchecker then you will get original fingerprint, use "yasnac" and "play integrity api checker"

**Comment by XdekHckr** [comment]
> Thanks for confirming, I suspected those associated values and for clarifying they pif.json values don't get show via getprops. Kinda a bummer there we don't know a tool or way to see the pif.json values applied!
> 
> vlad, from what you've said, I suppose I'll just have to go on faith and keep trying different "TheFreeman193" files 'til one works (tho the bummer is that it seems Google has some kind of "10k" 24 hour limit thingy, so I can only try a tiny few per day)

mine doesn't have any limit, also just get your own fingerprint from ROMs or older device (I got mine from old redmi) and you are good to go, the public ones from generators like thefreeman193 or xiaomi generator (from pixel flasher app) will be banned fast. 
https://www.youtube.com/watch?v=4Omyr_OR7eY

**Comment by XdekHckr** [comment]
btw have you tried using [play integrity next ](https://github.com/daboynb/PlayIntegrityNEXT) along with chiteroman's module? It would be easier and faster to get fingerprints working for your device (still not better than obtaining them manually, but easier)

**Comment by vladrevers** [comment]
> use "yasnac" and "play integrity api checker"

I have these apps also show the original device fingerprint, just like the Play Store via verification in "Developer options > Check integrity". The only clear difference between the working pif.json is that MEETS_DEVICE_INTEGRITY is also successfully checked, and without it (or with locked data), only MEETS_BASIC_INTEGRITY is successfully checked.

**Comment by markfm62** [comment]
Net, the MacabeeY issue appears to have been a non-PIF module that was breaking Device Integrity.  Once that module was replaced by an earlier version, they cleared Device Integrity.

**Comment by MaccabeeY** [comment]
@markfm62 really helped me out clear DEVICE, eternally grateful...

Also, the non-PIF module that was I using that @markfm62 was speaking of, was a specific version of @chiteroman BootloaderSpoofer, specifically v3.8.  When I replaced BootloaderSpoofer v3.8 with v1.2, DEVICE passed.
